libs/thunk/mock/mock.cpp

```diff
@@ -108,6 +108,78 @@ _close(int file_handle)
     return close_handler(file_handle);
 }
 
+bool
+_is_native_program(_In_ const char* file_name)
+{
+    std::string file_name_string(file_name);
+    std::string file_extension = file_name_string.substr(file_name_string.find_last_of(".") + 1);
+    if (file_extension == "dll") {
```

Comment:
should this also check for .sys?

Reply:
The mock implementation checks for .dll as for unit tests we will use .dll to load the native code.